### DISCUSSION TITLE
WAU not working

### DISCUSSION THREAD

**Discussion by alevalesantos** [Romanitho/Winget-AutoUpdate#791] [General]
Hi

I trying to install the new version of the Winget-AutoUpdate, I did this steps:

1 - Install the Winget-AutoUpdate-aaS over the store for business
2 - Install the Winget-AutoUpdate (WAU) using the .MSI
3 - Imported and setup the ADMX

I deployed the configuration over the Intune with sucess, but the windows task scheduler was not changed the execution time/date and I cannot see the white list file.

How can I confirm if everthing are working fine ?

Thanks

**Comment by AndrewDemski-ad-gmail-com** [comment]
Please check if you can find the correct options in system registry:
If all arguments passed to MSI were correct then those settings should be visible there

the location: 
>HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Romanitho\Winget-AutoUpdate
![image](https://github.com/user-attachments/assets/54383d20-b857-4ad7-8a39-c545e0d5f22c)


GL
AD

**Reply by alevalesantos** [reply]
I have the registry but I using the ADMX to change the settings.

I did this steps:
1 - Installed the WAU.MSI -silent
2 - Installed the WAU using Winget-AutoUpdate-Install.ps1
3 - Setup the ADMX files

I dont have the white list apps file, I added the 7zip.7zip but I didnt find the white list file created and I changed the "Updates Interval" to 12:00, and Windows Task still 06:00.
Im using the Intune, did I forget some step ?

**Reply by AndrewDemski-ad-gmail-com** [reply]
Hi @alevalesantos,
I could speculate that whatever is happening (or not) then it must log into error.txt file
https://github.com/Romanitho/Winget-AutoUpdate/blob/f4bcca43c3f14b186bfc4a5ce9121e67759c12a9/Sources/Winget-AutoUpdate/Winget-Upgrade.ps1#L256-L285

Both `Get-IncludedApps` and `Get-ExcludedApps` report errors to dedicated file and use `Write-ToLog` to report progress.

### an Idea
WAU is PowerShell 5.1 compliant, you may los it into PowerShell ISE and go codeblock-by-codeblock to find the missing element in your configuration.

It is the end of a very long pre-Xmas quarter of the year.
Hopefully you will find enough time to spend on that. 
It is difficult to guide you without hands-on experience with your configuration @alevalesantos.

Good Luck,
you'll find the problem and you'll probably laugh a little bit about it.
AD

**Comment by alevalesantos** [comment]
<img width="698" alt="image" src="https://github.com/user-attachments/assets/03b36bf6-76ac-4735-b04f-459424bdd7ec" />
<img width="692" alt="image" src="https://github.com/user-attachments/assets/99e784a2-1072-4848-be84-e042844f3399" />

**Comment by alevalesantos** [comment]
I guess I found.
Now the white list are managed by registry (I guess), but I cannot see the windows schedule with tasks to run at 11:00, only 06:00

Should I have the Windows Schedules tasks to run at 11:00 ?

**Reply by AndrewDemski-ad-gmail-com** [reply]
Yes, you are on track.
The [advanced installation](https://github.com/Romanitho/Winget-AutoUpdate/blob/main/README.md#advanced-installation) section of WAU readme states this:
>UPDATESATTIME
Default value 6AM (06:00:00). Specify the time of the update interval execution time. Example UPDATESATTIME="11:00:00"

Once WAU Tasks are generated using default value, you would need to edit them manually of use WAU MSI installer with that custom property to create them with correct time during Re/install.

Enjoy the weekend
AD

**Comment by alevalesantos** [comment]
<img width="697" alt="image" src="https://github.com/user-attachments/assets/8f64a9f0-ccb5-4b0d-b000-a64731362e8c" />
<img width="245" alt="image" src="https://github.com/user-attachments/assets/da7ebdb7-9262-4a78-83f9-0028ed3cba90" />
Ok, but not make sense, because the parameter "Updates at Time" should be responsadible to manage this settings.
I you dont have any setting to change the "update at time", the default time is 06:00, but if you have the parameter configurared, this setting should be respected.
I remember in the past this was working fine.

**Reply by AndrewDemski-ad-gmail-com** [reply]
As always the devils in the details.
The scheduled action is triggered by event, and event alone.
If there was an active service waiting or the right time, then it would launch WAU at the defined moment.
Task scheduler does not have that ability.

**Comment by Romanitho** [comment]
Don't confuse this with the fork Winget-AutoUpdate-aaS. We do not maintain it, and it may cause issues when installing WAU.

**Comment by DevRWC2023** [comment]
So i have uploaded the ADMX into Intune however i cannto find it under the Configuration Profile.
ANy help on this would be Great thank you. As i am pulling my hair out been looking through all settings

**Reply by Romanitho** [reply]
you should find all you need here:
https://learn.microsoft.com/en-us/mem/intune-service/configuration/administrative-templates-import-custom

**Reply by DevRWC2023** [reply]
Thank you for the Quick Repsonce apprecaite it i have located it.

**Comment by cace72** [comment]
I think installation of MSI does not take notice of registry settings under policies during installation. That would be really great to have that option or possibility. Otherwise you have to install MSI with the same settings as the registry has. not sure what does install wau-policies task to scheduler.
